src/main/java/graphql/annotations/DefaultTypeFunction.java

```diff
@@ -230,9 +230,11 @@ public GraphQLType apply(Class<?> aClass, AnnotatedType annotatedType) {
         @Override
         public GraphQLType apply(Class<?> aClass, AnnotatedType annotatedType) {
             GraphQLName name = aClass.getAnnotation(GraphQLName.class);
-            String typeName = name == null ? aClass.getSimpleName() : name.value();
-            if (processing.containsKey(typeName)) {
-                return processing.get(typeName);
+            String typeName = name == null ? aClass.getName() : name.value();
```

Comment:
I detected this thanks to running tests in parallel: after fixing the race condition, Class1 from two tests were getting mixed since the simple name was being used here.